src/clp_ffi_js/ir/StructuredIrStreamReader.cpp

```diff
@@ -0,0 +1,195 @@
+#include "StructuredIrStreamReader.hpp"
+
+#include <cstddef>
+#include <format>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <system_error>
+#include <utility>
+#include <vector>
+
+#include <clp/Array.hpp>
+#include <clp/ErrorCode.hpp>
+#include <clp/ffi/ir_stream/Deserializer.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/Value.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/TraceableException.hpp>
+#include <emscripten/em_asm.h>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/ClpFfiJsException.hpp>
+#include <clp_ffi_js/constants.hpp>
+#include <clp_ffi_js/ir/StreamReader.hpp>
+#include <clp_ffi_js/ir/StreamReaderDataContext.hpp>
+
+namespace {
+constexpr std::string_view cEmptyJsonStr{"{}"};
+constexpr std::string_view cLogLevelFilteringNotSupportedErrorMsg{
+        "Log level filtering is not yet supported in this reader."
+};
+constexpr std::string_view cReaderOptionsTimestampKey{"timestampKey"};
+}  // namespace
+
+namespace clp_ffi_js::ir {
+using clp::ir::four_byte_encoded_variable_t;
```

Comment:
```suggestion
namespace clp_ffi_js::ir {
namespace {
constexpr std::string_view cEmptyJsonStr{"{}"};
constexpr std::string_view cLogLevelFilteringNotSupportedErrorMsg{
        "Log level filtering is not yet supported in this reader."
};
constexpr std::string_view cReaderOptionsTimestampKey{"timestampKey"};
}  // namespace

using clp::ir::four_byte_encoded_variable_t;
```

We want to narrow the scope of the definitions as much as possible (as usual).

Reply:
Makes sense. didn't know we could do anonymous namespaces.